components/pd_client/src/lib.rs

```diff
@@ -1,4 +1,6 @@
 // Copyright 2016 TiKV Project Authors. Licensed under Apache-2.0.
+#[allow(unused_extern_crates)]
+extern crate tikv_alloc;
```

Comment:
If it's unused, why need to link to it explictily?

Reply:
According to the description of `tikv_alloc`:

> Writing `extern crate tikv_alloc;` will link it to jemalloc when appropriate. The TiKV library itself links to `tikv_alloc` to ensure that any binary linking to it will use jemalloc. With few exceptions, _every binary and project in the TiKV workspace should link (perhaps transitively) to tikv_alloc_. This is to ensure that tests and benchmarks run with the production allocator. In other words, binaries and projects that don't link to `tikv` should link to `tikv_alloc`.

Reply:
Given that features is not specified in `Cargo.toml`, I don't see the point to link it everywhere. /cc @brson

Reply:
What do you think? @brson @nrc

Reply:
I think that since we depend on crates which depend on tikv_alloc, then it is not necessary here. Of course that is somewhat fragile because if we end up not depending on any crate and our transitive dependency disappears, then we wouldn't get any warning. So for the sake of two lines of boilerplate I think it is good to have a direct dependency.